In [1]:
# 쥬피터노트북 셀 스타일 조절 
from IPython.core.display import display, HTML
display(HTML("<style>.container { font-weight: bold !important; }</style>"))
display(HTML("<style>.container { width: 100% !important; }</style>"))

# 경고 메세지 숨기기 
import warnings
warnings.filterwarnings(action='ignore')

In [2]:
import pandas as pd
import numpy as np
import os

In [3]:
# ls data

# Q1-1

```
아래의 파일을 이용하여 df_covid 데이타프레임을 생성하여라.
   
seoul-covid19_6_30_.csv    

```


# Q1-2
```
'연번' 컬럼은 삭제하고  '환자' 컬럼 값 기준으로 정렬한 후 '환자' 컬럼을 인덱스로 설정하여라. 
```



In [112]:
df_covid

확진일   거주지     여행력          접촉력         조치사항
환자                                                  
2      1.24.   강서구  중국 우한시        해외 접촉  국립중앙의료원(퇴원)
5      1.30.   중랑구  중국 우한시        해외 접촉    서울의료원(퇴원)
6      1.30.   종로구       -        #3 접촉  서울대학교병원(퇴원)
7      1.30.   마포구  중국 우한시        해외 접촉    서울의료원(퇴원)
9      1.31.   성북구       -        #5 접촉    서울의료원(퇴원)
...      ...   ...     ...          ...          ...
12819  6.30.   마포구       -    대전 꿈꾸는 교회         서남병원
12822  6.29.   의정부       -         확인 중       타시도 이관
12829  6.30.   관악구       -         확인 중        보라매병원
12830  6.30.  영등포구       -  영등포구 확진자 접촉            -
12831  6.30.  영등포구       -         확인 중      중앙대학교병원

[1321 rows x 5 columns]

# Q1-3
어느 지역에서 제일 많은 확진자가 발생했는지 1~5위 까지 출력하여라 (거주지 기준)

# Q1-4
확진자가 가장 많이 발생한 1, 2, 3위 지역의 데이타만 출력하여라 


# Q1-5
'확진일' 컬럼값을 아래와 같은 형태로 변경하여라
<br>
'1.24.' => '2020-1-24'


# Q1-6
'확진일' 컬럼을 시계열 데이타로 형태를 변경하여라 

In [125]:
df_covid.dtypes

확진일     datetime64[ns]
거주지             object
여행력             object
접촉력             object
조치사항            object
dtype: object

## Q1-7

3월에 발생한 확진자의 데이타만 출력하여라.

## Q1-8
주말에 발생한 확진자의 데이타만 출력하여라.

## Q1-9
'콜센터'와 관련된 확진자의 데이타를 출력하여라. 

확진일   거주지 여행력         접촉력        조치사항
환자                                                
7265  2020-03-08   노원구   -  구로구 콜센터 관련    서북병원(퇴원)
7401  2020-03-09   은평구   -  구로구 콜센터 관련  생활치료센터(퇴원)
7444  2020-03-09   강서구   -  구로구 콜센터 관련    서남병원(퇴원)
7461  2020-03-09    중구   -  구로구 콜센터 관련   서울의료원(퇴원)
7463  2020-03-09   금천구   -  구로구 콜센터 관련   서울의료원(퇴원)
...          ...   ...  ..         ...         ...
8937  2020-03-22  서대문구   -    콜센터직원 접촉    서북병원(퇴원)
8943  2020-03-22  서대문구   -    콜센터직원 접촉  생활치료센터(퇴원)
9170  2020-03-25   강서구   -    콜센터직원 접촉    서남병원(퇴원)
9867  2020-03-31   양천구   -    콜센터직원 접촉    서남병원(퇴원)
10009 2020-04-02   양천구   -    콜센터직원 접촉   보라매병원(퇴원)

[98 rows x 5 columns]

## Q1-10
'우한'과 관련된 확진자의 데이타만 출력하여라. 

확진일   거주지     여행력    접촉력         조치사항
환자                                             
2  2020-01-24   강서구  중국 우한시  해외 접촉  국립중앙의료원(퇴원)
5  2020-01-30   중랑구  중국 우한시  해외 접촉    서울의료원(퇴원)
7  2020-01-30   마포구  중국 우한시  해외 접촉    서울의료원(퇴원)
13 2020-02-02   인천시   우한 교민  해외 접촉  국립중앙의료원(퇴원)
23 2020-02-06  서대문구  중국 우한시  해외 접촉  국립중앙의료원(퇴원)
24 2020-02-07   아산시    우한교민  해외 접촉  국립중앙의료원(퇴원)

## Q1-11
사망자는 모두 몇명인가?

# Q2-1

아래의 파일을 이용하여 데이타프레임을 생성하여라.<br>
   
전국 평균 분양가격.csv 



# Q2-2

행과 열을 아래의 명령어를 이용하여 서로 변경하여라.  (df는 데이타프레임명)
```
df = df.T

```



0     1     2      3     4     5     6      7     8     9     10  \
지역           서울    부산    대구     인천    광주    대전    울산     경기    세종    강원    충북   
2013년12월  18189  8111  8080  10204  6098  8321  8090  10855  7601  6230  6589   
2014년1월   17925  8111  8080  10204  7326  8321  8090  10855  7600  6230  6589   
2014년2월   17925  9078  8077  10408  7611  8321  8090  10791  7532  6230  6611   
2014년3월   18016  8965  8101  10408  7346  8341  8153  10784  7814  6141  6625   
2014년4월   18098  9402  8267  10000  7346  8341  8153  10876  7908  6373  6678   
2014년5월   19446  9501  8274   9844  7523  8341  8153  10646  7934  6350  6598   
2014년6월   18867  9453  8360  10058  7659  8333  8153  10266  8067  6350  6587   
2014년7월   18742  9457  8360   9974  7612  8333  8153  10124  8067  6268  6586   
2014년8월   19274  9411  8370   9973  7622  8333  8493  10134  8141  6268  6586   
2014년9월   19404  9258  8449   9973  7802  8048  8493  10501  8282  6419  6584   
2014년10월  19759  9110  8403  10016  7707  8038  8627  10397  8527  6631  6529   
2014년11월  20242  9208  8439  10020  7752  8067  8891  10356  8592  6365  6724   
2014년12월  20269  9208  8253  10020  7748  8067  8891  10379  8560  6365  6743   
2015년1월   20670  9204  8327  10017  7752  8067  8526  10391  8560  6348  6749   
2015년2월   20670  9235  8416   9876  7756  8067  8526  10355  8560  6350  6747   
2015년3월   19415  9279  8441   9876  7861  8067  8629  10469  8555  6182  6783   
2015년4월   18842  9327  8446   9938  7914  8145  9380  10684  8546  6924  6790   
2015년5월   18367  9345  8568  10551  7877  8272  9192  10685  8546  6846  6805   
2015년6월   18374  9515  8542  10443  7881  8079  9190  10573  8671  6986  6682   
2015년7월   18152  9559  8542  10443  8089  8079  9190  10518  8669  7019  6601   
2015년8월   18443  9581  8795  10449  8231  8079  9215  10573  8695  7008  6603   

            11    12    13    14    15    16  
지역          충남    전북    전남    경북    경남    제주  
2013년12월  6365  6282  5678  6168  6473  7674  
2014년1월   6365  6281  5678  6168  6485  7900  
2014년2월   6379  5946  5678  6234  6502  7900  
2014년3월   6287  5966  5696  6317  6610  7900  
2014년4월   6552  6277  5736  6412  6599  7900  
2014년5월   6591  6306  5656  6409  6610  7900  
2014년6월   6644  6351  5609  6554  6615  7914  
2014년7월   6805  6319  5780  6556  6613  7914  
2014년8월   6914  6436  5685  6563  6606  7914  
2014년9월   6882  6719  5804  6577  6767  7833  
2014년10월  6831  6581  5753  6778  6881  7724  
2014년11월  6940  6583  5768  6881  7125  7724  
2014년12월  6989  6583  5784  6989  7332  7739  
2015년1월   6976  6583  5784  6992  7592  7739  
2015년2월   6980  6583  5833  6953  7588  7739  
2015년3월   7161  6542  5825  6997  7668  7826  
2015년4월   7017  6551  5940  7006  7683  7285  
2015년5월   6975  6556  6050  6966  7717  7285  
2015년6월   6939  6601  6243  6887  7715  7343  
2015년7월   6935  6750  6286  7035  7723  7343  
2015년8월   6942  6580  6289  7037  7665  7343

# Q2-3

1행을 삭제하고 2행을 컬럼으로 지정하여라. 

지역           서울    부산    대구     인천    광주    대전    울산     경기    세종    강원    충북  \
2013년12월  18189  8111  8080  10204  6098  8321  8090  10855  7601  6230  6589   
2014년1월   17925  8111  8080  10204  7326  8321  8090  10855  7600  6230  6589   
2014년2월   17925  9078  8077  10408  7611  8321  8090  10791  7532  6230  6611   
2014년3월   18016  8965  8101  10408  7346  8341  8153  10784  7814  6141  6625   
2014년4월   18098  9402  8267  10000  7346  8341  8153  10876  7908  6373  6678   
2014년5월   19446  9501  8274   9844  7523  8341  8153  10646  7934  6350  6598   
2014년6월   18867  9453  8360  10058  7659  8333  8153  10266  8067  6350  6587   
2014년7월   18742  9457  8360   9974  7612  8333  8153  10124  8067  6268  6586   
2014년8월   19274  9411  8370   9973  7622  8333  8493  10134  8141  6268  6586   
2014년9월   19404  9258  8449   9973  7802  8048  8493  10501  8282  6419  6584   
2014년10월  19759  9110  8403  10016  7707  8038  8627  10397  8527  6631  6529   
2014년11월  20242  9208  8439  10020  7752  8067  8891  10356  8592  6365  6724   
2014년12월  20269  9208  8253  10020  7748  8067  8891  10379  8560  6365  6743   
2015년1월   20670  9204  8327  10017  7752  8067  8526  10391  8560  6348  6749   
2015년2월   20670  9235  8416   9876  7756  8067  8526  10355  8560  6350  6747   
2015년3월   19415  9279  8441   9876  7861  8067  8629  10469  8555  6182  6783   
2015년4월   18842  9327  8446   9938  7914  8145  9380  10684  8546  6924  6790   
2015년5월   18367  9345  8568  10551  7877  8272  9192  10685  8546  6846  6805   
2015년6월   18374  9515  8542  10443  7881  8079  9190  10573  8671  6986  6682   
2015년7월   18152  9559  8542  10443  8089  8079  9190  10518  8669  7019  6601   
2015년8월   18443  9581  8795  10449  8231  8079  9215  10573  8695  7008  6603   

지역          충남    전북    전남    경북    경남    제주  
2013년12월  6365  6282  5678  6168  6473  7674  
2014년1월   6365  6281  5678  6168  6485  7900  
2014년2월   6379  5946  5678  6234  6502  7900  
2014년3월   6287  5966  5696  6317  6610  7900  
2014년4월   6552  6277  5736  6412  6599  7900  
2014년5월   6591  6306  5656  6409  6610  7900  
2014년6월   6644  6351  5609  6554  6615  7914  
2014년7월   6805  6319  5780  6556  6613  7914  
2014년8월   6914  6436  5685  6563  6606  7914  
2014년9월   6882  6719  5804  6577  6767  7833  
2014년10월  6831  6581  5753  6778  6881  7724  
2014년11월  6940  6583  5768  6881  7125  7724  
2014년12월  6989  6583  5784  6989  7332  7739  
2015년1월   6976  6583  5784  6992  7592  7739  
2015년2월   6980  6583  5833  6953  7588  7739  
2015년3월   7161  6542  5825  6997  7668  7826  
2015년4월   7017  6551  5940  7006  7683  7285  
2015년5월   6975  6556  6050  6966  7717  7285  
2015년6월   6939  6601  6243  6887  7715  7343  
2015년7월   6935  6750  6286  7035  7723  7343  
2015년8월   6942  6580  6289  7037  7665  7343

# Q2-4

'지역' 컬럼을 시계열 데이타로 변경하고 인덱스로 설정하여라. 

In [136]:
df.sample(3)

지역             서울    부산    대구     인천    광주    대전    울산     경기    세종    강원  \
2014-06-01  18867  9453  8360  10058  7659  8333  8153  10266  8067  6350   
2015-06-01  18374  9515  8542  10443  7881  8079  9190  10573  8671  6986   
2014-04-01  18098  9402  8267  10000  7346  8341  8153  10876  7908  6373   

지역            충북    충남    전북    전남    경북    경남    제주  
2014-06-01  6587  6644  6351  5609  6554  6615  7914  
2015-06-01  6682  6939  6601  6243  6887  7715  7343  
2014-04-01  6678  6552  6277  5736  6412  6599  7900

# Q2-5

2015년 6월 이후의 데이타만 출력하여라 

# Q2-6

2014년도의 서울, 인천 경기 컬럼만 출력하여라. 

지역             서울     인천     경기
2014-01-01  17925  10204  10855
2014-02-01  17925  10408  10791
2014-03-01  18016  10408  10784
2014-04-01  18098  10000  10876
2014-05-01  19446   9844  10646
2014-06-01  18867  10058  10266
2014-07-01  18742   9974  10124
2014-08-01  19274   9973  10134
2014-09-01  19404   9973  10501
2014-10-01  19759  10016  10397
2014-11-01  20242  10020  10356
2014-12-01  20269  10020  10379

# Q2-7
csv 파일 형태로 현재의 데이타프레임을 저장하여라. 
<br>
output/평균분양가격_서울인천경기.csv